web/twa-vis-platform/code/server.js

```diff
@@ -48,75 +49,109 @@ let store;
+            const { preferred_username: userName, given_name: firstName, family_name: lastName, name: fullName, realm_access: { roles }, resource_access: clientRoles } = req.kauth.grant.access_token.content;
+            res.json({ userName, firstName, lastName, fullName, roles, clientRoles });
         });
-      } catch (error) {
-        console.info('Error while creating Redis Client, please ensure that Redis is running and the host is specified as an environment variable if this viz app is in a Docker container');
-        console.error(error);
-      }
-      redisClient.connect().catch('Error while creating Redis Client, please ensure that Redis is running and the host is specified as an environment variable if this viz app is in a Docker container', console.error);
-      store = new RedisStore({
-        client: redisClient,
-        prefix: "redis",
-        ttl: undefined,
-      });
-    } else {
-      store = new MemoryStore(); // use in-memory store for session data in dev mode
-      console.info(`development mode is:`, dev ? colourYellow : colourRed, dev, colourReset, `-> using in-memory session store (express-session MemoryStore())`);
+
+        const protectedPages = process.env.PROTECTED_PAGES.split(',');
+        protectedPages.forEach(page => {
+            server.get(page, keycloak.protect());
+        });
+        const roleProtectedPages = process.env.ROLE_PROTECTED_PAGES.split(',');
+        roleProtectedPages.forEach(page => {
+            server.get(page, keycloak.protect(process.env.ROLE));
+            console.info('protecting page', page, 'with role', process.env.ROLE);
+        });
+
+        const useGeoServerProxy = process.env.REACT_APP_USE_GEOSERVER_PROXY === 'true';
+        console.log('REACT_APP_USE_GEOSERVER_PROXY is ' + useGeoServerProxy);
+
+        if (useGeoServerProxy) {
+            console.log('GeoServer requests from MapBox will be sent to /geoserver-proxy')
+        }
+
+        if (useGeoServerProxy) {
+            server.get('/geoserver-proxy', async (req, res) => {
+                const targetUrl = req.query.url;
+                let headers = { ...req.headers };
+
+                if (req.kauth?.grant) {
+                    headers['Authorization'] = 'Bearer ' + req.kauth.grant.access_token.token;
+                }
+
+                try {
+                    // Forward the request to the target URL with the modified headers
+                    const response = await axios({
+                        url: targetUrl,
+                        method: req.method,
+                        headers: headers,
+                        responseType: 'stream', // To stream the response back
+                    });
+
+                    // Pipe the response back to the client
+                    response.data.pipe(res);
+                } catch (err) {
+                    // most of these errors can probably be ignored
+                    console.error(err);
+                }
```

Comment:
I think we should get rid of this whole path and just save the `req.kauth.grant.access_token` to the already existing /api/userinfo/path or somewhere else as a global variable. Then call that directly from the mapbox container url transformer

Reply:
The intention behind this design is that the token needs to be taken directly from the session store/server for each request, otherwise the token gets outdated and becomes invalid.

Reply:
I think this design guarantees that each request is handled by the middleware